src/domain/email/email.repository.ts

```diff
@@ -19,24 +44,22 @@ export class EmailRepository implements IEmailRepository {
     account: string;
   }): Promise<void> {
     const email = new EmailAddress(args.emailAddress);
-    const verificationCode = codeGenerator();
-
-    // Pads the final verification code to 6 characters
-    // The generated code might have less than 6 digits so the version to be
-    // validated against should account with the leading zeroes
-    const paddedVerificationCode = verificationCode.toString().padStart(6, '0');
+    const verificationCode = this._generateCode();
 
     try {
       await this.emailDataSource.saveEmail({
         chainId: args.chainId,
-        code: paddedVerificationCode,
+        code: verificationCode,
         emailAddress: email,
         safeAddress: args.safeAddress,
         signer: args.account,
         codeGenerationDate: new Date(),
       });
-
-      // TODO if successful, send the generated code (result.verification)
+      await this._sendEmailVerification({
+        ...args,
+        signer: args.account,
```

Comment:
Out of the scope of this PR but to take note: we should rename either `signer` to `account` or vice versa for consistency.